app/vmctl/native/client.go

```diff
@@ -35,39 +41,90 @@ type Response struct {
 
 // Explore finds metric names by provided filter from api/v1/label/__name__/values
 func (c *Client) Explore(ctx context.Context, f Filter, tenantID string) ([]string, error) {
-	url := fmt.Sprintf("%s/%s", c.Addr, nativeMetricNamesAddr)
-	if tenantID != "" {
-		url = fmt.Sprintf("%s/select/%s/prometheus/%s", c.Addr, tenantID, nativeMetricNamesAddr)
-	}
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
-	if err != nil {
-		return nil, fmt.Errorf("cannot create request to %q: %s", url, err)
+	exploreChunk := f.Chunk
+	if exploreChunk == stepper.StepHour ||
+		exploreChunk == stepper.StepMinute ||
+		exploreChunk == stepper.StepDay ||
+		exploreChunk == "" {
+		// the minimal step wil be used for metrics explore process
+		exploreChunk = stepper.StepWeek
 	}
 
-	params := req.URL.Query()
-	if f.TimeStart != "" {
-		params.Set("start", f.TimeStart)
+	start, err := utils.GetTime(f.TimeStart)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse time start for explore metrics: %s", err)
 	}
-	if f.TimeEnd != "" {
-		params.Set("end", f.TimeEnd)
+	end, err := utils.GetTime(f.TimeEnd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse time end for explore metrics: %s", err)
 	}
-	params.Set("match[]", f.Match)
-	req.URL.RawQuery = params.Encode()
 
-	resp, err := c.do(req, http.StatusOK)
+	ranges, err := stepper.SplitDateRange(start, end, exploreChunk, false)
 	if err != nil {
-		return nil, fmt.Errorf("series request failed: %s", err)
-	}
 
-	var response Response
-	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return nil, fmt.Errorf("cannot decode series response: %s", err)
-	}
+		return nil, fmt.Errorf("failed to create date ranges for explore metrics: %w", err)
+	}
+
+	var metricNames []string
+	errs, ctx := errgroup.WithContext(ctx)
+	metricNamesC := make(chan []string)
+	for _, times := range ranges {
+		start := times[0].Format(time.RFC3339)
+		end := times[1].Format(time.RFC3339)
+		errs.Go(func() error {
```

Comment:
the concurrency of the series discovery should be limited by vm-concurrency flag

Reply:
limiter was added